proxy_agent/src/shared_state.rs

```diff
@@ -108,13 +117,73 @@ impl Default for SharedState {
             telemetry_reader_shutdown: false,
             telemetry_logger_shutdown: false,
             telemetry_logger_status_message: UNKNOWN_STATUS_MESSAGE.to_string(),
+            mock_server_ip: None,   // test only
```

Comment:
I'd recommend not putting test things here, but if you absolutely have to you can put them behind `#cfg(test)`